tests/unit/lms/services/oauth_http_test.py

```diff
@@ -36,24 +37,104 @@ def test_convenience_methods(self, method, svc, http_service, oauth2_token_servi
         )
         assert response == http_service.request.return_value
 
-    def test_it_crashes_if_theres_already_an_Authorization_header(self, svc):
+    def test_request_crashes_if_theres_already_an_Authorization_header(self, svc):
         with pytest.raises(AssertionError):
             svc.request(sentinel.method, sentinel.url, headers={"Authorization": "foo"})
 
-    def test_it_raises_if_theres_no_access_token_for_the_user(
+    def test_request_raises_if_theres_no_access_token_for_the_user(
         self, svc, oauth2_token_service
     ):
         oauth2_token_service.get.side_effect = OAuth2TokenError
 
         with pytest.raises(OAuth2TokenError):
             svc.request(sentinel.method, sentinel.url)
 
-    def test_it_raises_if_the_HTTP_request_fails(self, svc, http_service):
+    def test_request_raises_if_the_HTTP_request_fails(self, svc, http_service):
         http_service.request.side_effect = HTTPError
 
         with pytest.raises(HTTPError):
             svc.request(sentinel.method, sentinel.url)
 
+    def test_access_token_request(
+        self,
+        svc,
+        http_service,
+        OAuthTokenResponseSchema,
+        oauth_token_response_schema,
+        oauth2_token_service,
+    ):
+        oauth_token_response_schema.parse.return_value = {
+            "access_token": "access_token",
+            "refresh_token": "refresh_token",
+            "expires_in": 1234,
+        }
```

Comment:
This might be better as part of the fixture, then later on, instead of having matching values we'd say:

```python
    response = oauth_token_response_schema.parse.return_value
    oauth2_token_service.save.assert_called_once_with(
        response["access_token"], response["refresh_token"], response["expires_in"]
    )
```

Which would make this test a little shorter and more explicit about where the values some from.

Reply:
I didn't move it into the fixture as it's only used by this one test, but I put it in a variable: https://github.com/hypothesis/lms/pull/2987/commits/e1715f11fb313de4da2d31223d2792e9a5d9b4e6

Reply:
Yep, that works too